File: java/271A.java
// 271 A. Beautiful Year

import java.util.*;

class Main {
    static boolean isBeautiful(int year) {
        String s = String.valueOf(year);
        Set<Character> set = new HashSet<>();
        for (char c : s.toCharArray()) {
            set.add(c);
        }
        return set.size() == s.length();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int y = sc.nextInt();

        while(true){
            y++;
            if(isBeautiful(y)) {
                System.out.println(y);
                break;
            }
        }
        sc.close();
    }
}

File: cpp/271A.cpp
// 271 A. Beautiful Year

#include <iostream>
#include <set>
using namespace std;

bool isBeautiful(int year) {
    string s = to_string(year);
    set<char> st(s.begin(), s.end());
    return st.size() == s.size();
}

int main() {
    int y;
    cin >> y;
    while (true) {
        y++;
        if (isBeautiful(y)) {
            cout << y;
            break;
        }
    }
    return 0;
}